FlexNetworking/FlexNetworking.swift

```diff
@@ -268,3 +290,102 @@ public class FlexNetworking {
 
     internal lazy var _rx = Rx(flex: self)
 }
+
+extension FlexNetworking: URLSessionDataDelegate, URLSessionDownloadDelegate {
+
+    public func urlSession(_ session: URLSession, downloadTask: URLSessionDownloadTask, didWriteData bytesWritten: Int64, totalBytesWritten: Int64, totalBytesExpectedToWrite: Int64) {
+        guard let taskID = downloadTask.taskDescription else {
+            return
+        }
+
+        self.progressObservers[taskID]?(Float(totalBytesWritten) / Float(totalBytesExpectedToWrite))
+    }
+
+    public func urlSession(_ session: URLSession, dataTask: URLSessionDataTask, didReceive response: URLResponse, completionHandler: @escaping (URLSession.ResponseDisposition) -> Void) {
+        if let taskID = dataTask.taskDescription, response.expectedContentLength != SwiftNSURLResponseUnknownLength {
+            self.expectedDataLength[taskID] = response.expectedContentLength
+        }
+        completionHandler(.allow)
+    }
+
+    public func urlSession(_ session: URLSession, dataTask: URLSessionDataTask, didReceive data: Data) {
+        guard let taskID = dataTask.taskDescription else {
+            return
+        }
+
+        if self.dataSoFar[taskID] == nil {
+            self.dataSoFar[taskID] = Data()
+        }
+        self.dataSoFar[taskID]?.append(data)
+        let progress: Float
+        if let expectedLength = self.expectedDataLength[taskID] {
+            progress = Float(self.dataSoFar[taskID]?.count ?? 0) / Float(expectedLength)
+        } else {
+            progress = -1
+        }
```

Comment:
if we don't have an expectedDataLength and therefore don't know what the progress is, what's better: -1 as a missing value or a nullable progress value

Reply:
-1 is fine for now